### PR TITLE
PA-68

### PR DESCRIPTION
**Pull request by vasilistefanenko**
Add boot layers

### CODE REVIEW

File: infrastructure/infrastructure.go [hunk 1]
```diff
@@ -1,70 +1,70 @@
...
-			Publisher:  chn,
-			Subscriber: chn,
-		}
-
-	*/
+func Boot(ctx context.Context) (<-chan struct{}, error) {
 	var (
-		queue  PubSub
-		rawCfg = kafka.DefaultSaramaSubscriberConfig()
+		done = make(chan struct{})
```
Comment (MikeKonan) [inline infrastructure/infrastructure.go]: done = make(chan struct{}, 1)

File: domain/domain.go [hunk 2]
```diff
@@ -2,16 +2,34 @@ package domain
...
 var svc *sagaService
 
+type SagaServiceConfig struct {
+	SagasDir string `config:"sagas-dir,required"`
+}
+
 // Prepares domain layer
-func Boot() (err error) {
+func Boot(ctx context.Context) (<-chan struct{}, error) {
+	var done = make(chan struct{})
```
Comment (MikeKonan) [inline domain/domain.go]: limit 1

File: domain/domain.go [hunk 3]
```diff
@@ -2,16 +2,34 @@ package domain
...
 	"github.com/ThreeDotsLabs/watermill/message"
+	"gopkg.in/yaml.v2"
+
+	"github.com/HnH/choreographer/infrastructure"
 )
 
 var svc *sagaService
 
+type SagaServiceConfig struct {
+	SagasDir string `config:"sagas-dir,required"`
```
Comment (HnH) [inline domain/domain.go]: Naming. It's already saga service cfg, so SagasDir can be just Dir.

File: domain/domain.go [hunk 4]
```diff
@@ -33,7 +51,38 @@ func Boot() (err error) {
 	//svc.repo.routineCtx.ctx, svc.repo.routineCtx.done = context.WithCancel(context.Background())
 	go svc.repo.routine()
 
-	return
+	err := filepath.Walk(cfg.SagasDir, func(path string, f os.FileInfo, _ error) (err error) {
+		if f.IsDir() || filepath.Ext(f.Name()) != ".yml" {
```
Comment (HnH) [inline domain/domain.go]: Can we add file extension to cfg, with ".yml" as default?

File: domain/domain.go [hunk 5]
```diff
@@ -33,7 +51,38 @@ func Boot() (err error) {
...
+		}
+
+		var dto SagaDto
+		if err = yaml.NewDecoder(r).Decode(&dto); err != nil {
+			return
+		}
+
+		return svc.Register(dto)
+	})
+	if err != nil {
```
Comment (HnH) [inline domain/domain.go]: blank line

**Pull request merged**